docs/design/configs.md

```diff
@@ -0,0 +1,130 @@
+# Generic Config Object
+
+## Abstract
+
+A generic ``Config`` object to store command-line flags and environment variables
+that is used for configuration of ``Kubernetes`` components living on top of ``etcd``
+is proposed in this proposal.
```

Comment:
* reduced operational burden -- don't have to run your own etcd
* simplified deployment orchestration -- don't have to launch etcd, wait for it to become ready, write config to etcd, then launch app
* compatibility with k8s tooling (e.g., kubectl create -f, labels, etc.)
* use Kubernetes auth and security policy
* GC config when consumers are deleted -- if we implement existence dependencies as proposed in #1535
* consume via file (if we support a volume, as with secrets) or env vars (if we support that) rather than via API
* for Kubernetes components, preserves abstraction of the underlying storage mechanism (e.g., Kubelet doesn't need to interact directly with etcd)